website/source/docs/builders/openstack.html.md

```diff
@@ -157,6 +157,41 @@ builder.
 -   `security_groups` (array of strings) - A list of security groups by name to
     add to this instance.
 
+-   `source_image_filter` (object) - Filters used to populate filter options.
+    Example:
+
+    ``` json
+    {
+        "source_image_filter": {
+            "filters": {
+                "name": "ubuntu-16.04*",
+                "visibility": "protected",
+                "owner": "d1a588cf4b0743344508dc145649372d1",
+                "tag": "prod"
+            },
+            "most_recent": true
+        }
+    }
+    ```
+
+    This selects the most recent production Ubuntu 16.04 shared to you by the given owner.
+    NOTE: This will fail unless *exactly* one image is returned, or `most_recent` is set to true.
+    In the example, `most_recent` will cause this to succeed by selecting the newest image.
+
+    -   `filters` (map of strings) - filters used to select a `source_image`.
+        NOTE: This will fail unless *exactly* one image is returned, or `most_recent` is set to true.
+        Any filter described in the docs for [ImageService](https://developer.openstack.org/api-ref/image/v2/)
+        is valid.
```

Comment:
I think we should limit the filters we support to:
- name
- owner
- protected
- tags
- visibility
- memeber_status (I'm not sure what this is?)

The rest doesn't make any sense to expose.

Reply:
So I went through the ListOpts and imageservice/v2 documentation and here are my thoughts

I don't believe the following filters make sense in the context of our needs:

- status (has to be available, why allow it to be specified at all if there's only one value)
- marker (involved when doing subsequent requests, not relevant)
- member_status (must be "accepted" to retrieve an image, no way you can be anything else and use the listed image)

The old sort method can be discarded as well. The newer sort allows multiple keys and is documented well under [v2/images](https://developer.openstack.org/api-ref/image/v2/). The docs explicitly state how multiple fields can be used in sorting with `sort`, but not with `sort_key` and `sort_dir`. I would suggest using the former and dismissing the latter if we're cutting down on applicable fields. 

The following should be included as well, and with an example use case provided:

- sort (e.g. allows configs to choose preference for smaller size image)
- limit (e.g. enables sort functionality to be used to its fullest without forcing users into most_recent)
- updated_at/created_at (e.g. only use images generated this sprint)
- size_min/size_max (e.g. don't overutilize server space)

I believe these fields would be directly beneficial to teams and although not every team will use them, removing the capabilities of these filters would be a drawback. I would be opening an issue and requesting at least two of these myself if they weren't included when I came across the `source_image_filter` feature. Furthermore, the fields that can be used by the tangential `source_ami_filter` feature for the AWS Instance builder allows even more fields to be used, so why reduce the functionality that the OpenStack gophercloud API allows us. In fact, the AWS implementation has the same approach as mine, which is to allow any and all filters described in its [DescribeImages](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html) be used as filters in `source_ami_filter` [source](https://www.packer.io/docs/builders/amazon-instance.html#source_ami_filter). 

All your suggested fields also make sense, and I agree they should be included as well.